src/main/python/ayab/engine/engine.py

```diff
@@ -126,7 +131,7 @@ def knit_config(self, image):
         # validate configuration options
         valid, msg = self.validate()
         if not valid:
-            self.emit_popup(msg)
+            self.emit_popup(cast(str,msg))
```

Comment:
technically the typing is tight enough that it _should_ know that `msg` is a string here- but it seems mypy isn't as fancy as the lexer in typescript-land!